src/interactions/slashCommands/mixins/CommandOptionWithChoices.ts

```diff
@@ -31,13 +31,24 @@ export abstract class ApplicationCommandOptionWithChoicesBase<T extends string |
 		// Validate name
 		ow(
 			name,
-			`${this.type === ApplicationCommandOptionType.String ? 'string' : 'integer'} choice name`,
+			`${
+				this.type === ApplicationCommandOptionType.String
+					? 'string'
+					: this.type === ApplicationCommandOptionType.Integer
+					? 'integer'
+					: 'number'
```

Comment:
I'd suggest making an enum to convert raw types to string names, like the following:

```ts
const ApplicationCommandOptionTypeNames = {
	[ApplicationCommandOptionType.String]: 'string',
	[ApplicationCommandOptionType.Integer]: 'integer',
	[ApplicationCommandOptionType.Number]: 'number'
} as const;
```

So then we can do:

```ts
ow(name, `${ApplicationCommandOptionTypeNames[this.type]} choice name`, stringPredicate);
```

Reply:
Same applies for the rest of the code.